From: uazo <uazo@users.noreply.github.com>
Date: Sat, 28 Jun 2025 08:17:07 +0000
Subject: Disable Viewport Segments

Disable Viewport Segments API, specifically for foldable devices,
since it exposes the geometry of the fold.

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 .../blink/common/features_cc/Disable-ViewportSegments.inc      | 1 +
 .../blink/renderer/platform/runtime_enabled_features.json5     | 3 +--
 2 files changed, 2 insertions(+), 2 deletions(-)
 create mode 100644 cromite_flags/third_party/blink/common/features_cc/Disable-ViewportSegments.inc

diff --git a/cromite_flags/third_party/blink/common/features_cc/Disable-ViewportSegments.inc b/cromite_flags/third_party/blink/common/features_cc/Disable-ViewportSegments.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/third_party/blink/common/features_cc/Disable-ViewportSegments.inc
@@ -0,0 +1 @@
+SET_CROMITE_FEATURE_DISABLED(kViewportSegments);
diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -5200,8 +5200,7 @@
       status: "stable",
     },
     {
-      name: "ViewportSegments",
-      status: "stable",
+      name: "ViewportSegments", status: "test", public: true
     },
     {
       name: "ViewTransitionLongCallbackTimeoutForTesting",
--
